### ISSUE TITLE
Drasil's expression language should have different kinds of multiplication 

### ISSUE THREAD

**Issue by bmaclach** [JacquesCarette/Drasil#2212]
A mutliplication between numbers of the same type should be represented in Drasil differently than a multiplication between numbers of different types (i.e. an integer times a real number). This distinction should be added to `Expr`, probably to the `ArithOper` type.

See related discussion starting with this comment: https://github.com/JacquesCarette/Drasil/commit/a40c9a32df0af722bc9a0b07cda9bdb584633856#commitcomment-40209185

**Comment by balacij** [comment]
[Swift's handbook](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html#ID326) mentions that it's for all operations. For now, I've started with separating the integer and "real" addition and multiplication.

I've started work on the [addMulSeparation](https://github.com/JacquesCarette/Drasil/tree/addMulSeparation) branch. Right now, it compiles but breaks stable due to some issues with the ordering in binding, so I'm just slowly fixing them.
